pulpcore/pulpcore/app/viewsets/base.py

```diff
@@ -158,3 +162,58 @@ class CreateDestroyReadNamedModelViewSet(mixins.CreateModelMixin,
 
     """
     pass
+
+
+class AsyncUpdateMixin(object):
+    """
+    Provides an update method that dispatches a task with reservation for a repository
+    """
+    def update(self, request, pk, **kwargs):
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        app_label = instance._meta.app_label
+        async_result = tasks.base.general_update.apply_async_with_reservation(
+            instance._meta.db_table, pk,
+            args=(pk, app_label, serializer.__class__.__name__),
+            kwargs={'data': request.data, 'partial': partial}
+        )
+        return OperationPostponedResponse([async_result], request)
+
+    def partial_update(self, request, *args, **kwargs):
+        kwargs['partial'] = True
+        return self.update(request, *args, **kwargs)
+
+
+class AsyncRemoveMixin(object):
+    """
+    Provides a delete method that dispatches a task with reservation for a repository
+    """
+    def destroy(self, request, pk, **kwargs):
+        """
+        Delete a model instance
+        """
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+        app_label = instance._meta.app_label
+        async_result = tasks.base.general_delete.apply_async_with_reservation(
+            instance._meta.db_table, pk,
+            args=(pk, app_label, serializer.__class__.__name__)
+        )
+        return OperationPostponedResponse([async_result], request)
+
+
+class CreateReadAsyncUpdateDestroyNamedModelViewset(mixins.CreateModelMixin,
```

Comment:
I strongly prefer (up to you though) that the models inherit from their mixins directly rather than creating these base ViewSets.